### DISCUSSION TITLE
Exchange data in CoSim

### DISCUSSION THREAD

**Discussion by mnabideltares** [KratosMultiphysics/Kratos#11237] [General]
Hi,

I am trying to exchange data between two solvers in CoSim. Those solvers are both based in GeoMechanicsApplication, Pressure and Thermal.

I need the water flux from the pressure solver (to Thermal solver). The Pressure is calculated on the nodes, while the fluxes are defined on the integration points. I need to exchange the fluxes. But when I try, it tells me that the variable is not on nodes.

Is it possible to exchange variables, based on integration points, in CoSim? Or CoSim just works with variables based on the solution nodes?

It is possible to exchange Material properties (double scalars) by CoSim?

Thank you for help.
Regards

**Comment by loumalouomega** [comment]
@philbucher FYI

Huuum, I don't know exactly for non-nodal values if there is anything designed for it. A priori with CoSimIO it is possible, my question if there is anything designed out of the box.

**Reply by philbucher** [reply]
CoSimIO is only required for coupling with external tools, it is not relevant here

**Comment by philbucher** [comment]
So far the CoSimApp does not work with any integration point quantities at all, since it was not clear to me how those could be treated uniformly (basically differs a lot in how the elements treat them)

Before we go into detail, the easiest would be to calculate the quantities on the nodes (historical and non-historical). This would work out of the box. Is this an option?

**Reply by mnabideltares** [reply]
@loumalouomega  and @philbucher 

Then in this case, I can read the pressure on the nodes and re-calculate the fluxes on the integration points.
Although it is not a desirable solution and not efficient, but at least it works.
And for Material properties, copy the required properties on both Material input files.
Thank you for help.

**Comment by mnabideltares** [comment]
This issue is solved by exchanging the node values and recalculate the fluxes on the integration points
